### DISCUSSION TITLE
how to pass argument to specific package build

### DISCUSSION THREAD

**Discussion by dante01yoon** [vercel/turborepo#5177] [Help] [answered]
Hello, my package @app use @log package as a dependencies.

I wan to run in Docker with argument name `mode`.
```Dockerfile
RUN pnpm turbo run build --filter=@app -- --mode production
```

the problem is --mode arguments passed to all dependencies with build command. how to force give --mode argument only to @app, but not to other packages?

**Comment by anthonyshew** [comment, marked as answer]
You'll want to put the flag onto the script of your `build` command in the package.json of your `app`. That way, you don't have to give it to `turbo` at all.

**Comment by juanporley** [comment]
I think the accepted answer is working around the problem. If I want to potentially pass two different values for this argument to the build script of a particular package, there's no way to achieve it. If I hardcode the argument in different scripts then what's the point of the argument if I can't use it through `turbo run build`?
